apps/system/test/unit/task_manager_test.js

```diff
@@ -1,1377 +1,783 @@
-        this.sinon.clock.tick();
-      });
+    test('Should emit "cardviewbeforeshow" and "cardviewshown"', (done) => {
+      var spyCardViewBeforeShow = spyEvent(window, 'cardviewbeforeshow');
+      var spyCardViewShown = spyEvent(window, 'cardviewshown');
+      tm.hide().then(() => {
+        var show = tm.show();
+        assert.isTrue(spyCardViewBeforeShow.called);
+        clock.tick(1000);
+        return show;
+      }).then(() => {
+        assert.isTrue(spyCardViewShown.called);
+      }).then(done, done);
+      clock.tick(1000);
+    });
 
-      test('cardsview should be active', function() {
-        assert.isTrue(cardsView.classList.contains('active'));
-        assert.isTrue(taskManager.isShown());
+    test('Should not show if already showing', (done) => {
+      var spyBeforeShow = spyEvent(window, 'cardviewbeforeshow');
+      assert.isTrue(tm.isShown()); // (noting that we're already shown)
+      tm.show().then(() => {
+        assert.isFalse(spyBeforeShow.called);
+        done();
       });
     });
 
-    suite('display cardsview via holdhome > when the keyboard is displayed',
-    function() {
-      setup(function(done) {
-        MockService.mockQueryWith('keyboardEnabled', true);
-        assert.isFalse(taskManager.isShown(), 'taskManager isnt showing yet');
-
-        MockStackManager.mStack = [
-          apps.sms
-        ];
-        MockStackManager.mCurrent = MockStackManager.mStack.length -1;
-        waitForEvent(window, 'cardviewshown')
-          .then(function() { done(); }, failOnReject);
-
-        sendHoldhome();
-
-        window.dispatchEvent(new CustomEvent('appclosed'));
-        this.sinon.clock.tick();
-        assert.isFalse(taskManager.isShown());
+    test('Should not respond to holdhome if already showing', () => {
+      var showSpy = sinon.spy(tm, 'show');
+      tm.respondToHierarchyEvent(new CustomEvent('holdhome'));
+      assert.isFalse(showSpy.called);
+    });
 
-        window.dispatchEvent(new CustomEvent('keyboardhidden'));
-        this.sinon.clock.tick();
-      });
+    test('Should open in response to holdhome from homescreen', (done) => {
+      clock.restore();
+      tm.hide().then(() => {
+        MockService.mockQueryWith(
+          'AppWindowManager.getActiveWindow', homescreenApp);
+        var showSpy = sinon.spy(tm, 'show');
 
-      test('cardsview should be active', function() {
-        assert.isTrue(cardsView.classList.contains('active'));
-        assert.isTrue(taskManager.isShown());
-      });
-    });
+        sinon.spy(homescreenApp, 'close');
 
-    suite('populated task manager in rocketbar >', function() {
-      setup(function() {
-        MockStackManager.mStack = [
-          apps.sms
-        ];
-        MockStackManager.mCurrent = MockStackManager.mStack.length -1;
-        showTaskManager(this.sinon.clock);
-      });
+        var handleShown = () => {
+          window.removeEventListener('cardviewshown', handleShown);
+          try {
```

Comment:
Ah, I didn't realize that mocha was smart enough to catch the appropriate uncaught exception in a test. I guess it's not necessary after all.